plugins/transport-nio/src/main/java/org/elasticsearch/http/nio/NettyListener.java

```diff
@@ -203,20 +203,20 @@ public ChannelPromise unvoid() {
     }
 
     public static NettyListener fromBiConsumer(BiConsumer<Void, Exception> biConsumer, Channel channel) {
-        if (biConsumer instanceof NettyListener) {
-            return (NettyListener) biConsumer;
+        if (biConsumer instanceof NettyListener nettyListener) {
+            return nettyListener;
         } else {
             ChannelPromise channelPromise = channel.newPromise();
             channelPromise.addListener(f -> {
                 Throwable cause = f.cause();
                 if (cause == null) {
                     biConsumer.accept(null, null);
                 } else {
-                    if (cause instanceof Error) {
+                    if (cause instanceof Exception exception) {
+                        biConsumer.accept(null, exception);
+                    } else {
                         ExceptionsHelper.maybeDieOnAnotherThread(cause);
                         biConsumer.accept(null, new Exception(cause));
-                    } else {
-                        biConsumer.accept(null, (Exception) cause);
                     }
```

Comment:
These two should be equivalent, but they are not quite the same. I'd suggest asking for a second reviewer for this type of code change. Personally, I wouldn't change the original, but I'd be happy to hear other opinions

Reply:
I was also a bit concerned about this change, I described my thinking processes here: https://github.com/elastic/elasticsearch/pull/82161#discussion_r776976224. Do you know who I could ask for a second opinion? I am pretty new here so I do not know who could be a good fit.

Reply:
I've added @tbrooks8 as a reviewer, since he's the author of the code :-).
@tbrooks8 we'd appreciate your help on this one, thank you.

Reply:
Thank you!